server.go

```diff
@@ -423,18 +433,27 @@ func (srv *Server) Shutdown() error {
 		return &Error{err: "server not started"}
 	}
 	srv.started = false
-	srv.lock.Unlock()
+	defer srv.lock.Unlock() // to make sure re-serve works do after shutdown
 
 	if srv.PacketConn != nil {
 		srv.PacketConn.Close()
 	}
 	if srv.Listener != nil {
+		// only TCP listener needs shutdown signal
+		srv.shutdownSignal <- true
 		srv.Listener.Close()
 	}
 
 	fin := make(chan bool)
 	go func() {
-		srv.inFlight.Wait()
+		// srv.inFlight.Wait()
+		var count uint32
+		for {
+			count = atomic.LoadUint32(&srv.inFlight)
```

Comment:
this won't work - it will busy wait and consume 100% cpu while doing it

Reply:
ah, god...my fault